source/develop/release-management/features/gluster-self-heal-monitoring.html.md

```diff
@@ -41,21 +41,51 @@ Host Maintenance and Fencing will be enabled/disabled based on the following cri
 | No Unsynced Entries | Quorum available without this Node | Allowed |
 | No Unsynced Entries | Quorum not available without this Node | Not Allowed |
 
+ More info about standard host fencing is available at http://old.ovirt.org/Automatic_Fencing#Automatic_Fencing http://old.ovirt.org/Fence_kdump https://www.youtube.com/watch?v=V1JQtmdleaM
+
 ## Entity Changes
 Following entities will be changed as part of Gluster self heal monitoring.
 
 ### Gluster_volume_bricks
   Following columns will be added to gluster_volume_bricks table.
   unsynced_entries  - integer - No.of unsynced entries in the brick.
   unsynced_entires_history - text - History of unsynced entries in the brick. It will a list of comma separated values.
-  
-### Gluster_volumes_view
-‘heal_status’ column will be be added to gluster_volumes_view. It will be computed based on unsynced_entries in the brick. If there is any unsynced_entry in any of the bricks in the volume then ‘heal_status’ will be ‘Needs_Healing’.
 
+## Change in BLL Commands
+###VdsNotRespondingTreatmentCommand (for Host fencing)
+   VdsNotRespondingTreatmentCommand is the central BLL command which takes care of varios host fencing related activites. This will be changed to consider gluster self-heal information during host fencing. shouldFencingBeSkipped method will be changed and following logic will be added.
+   
+   if (host supports both gluster & virt services)
+      if(all bricks are down)
+        allow host fencing
+      else(some || all bricks are up)
+        if (unsync'd entries  > 0 for any up brick) //Entris needs to be healed from this host
+          skip host fencing
+        else
+          if (quorum is maintained (bricks and nodes for ALL volumes) with this host down)
+            allow host fencing
+          else
+            skip host fencing
+
```

Comment:
> For 3, This is the most critical point. Either we have to check the host(gluster host) to get the latest value, or wait for max 
> 10 minutes. 'gluster volume heal info' adds some overhead on the gluster processes. So I don't think we can increase 
> the frequency further. If we know for sure that the bricks are down then we can allow the Power Management. I see two 
> solutions for this.
>
>    Run the 'gluster volume heal info' command to get the latest values before fencing.

Well, checking healing info on fence proxy host same as we currently check sanlock lease would be the most reliable way (assuming that getting volume healing info is not too time consuming). Also the option to check or not healing info can be passed from engine to VDSM as a fencing policy option same as in sanlock lease case.

>   Delay the power management for 10 mins.

I wouldn't go that way, because with 10 minutes delay we will break HA VMs contract (HA VMs cannot be restarted on different host until PM stop was successfully executed on host)